src/modules/communication/internals/send-grid-service.ts

```diff
@@ -0,0 +1,32 @@
+import { SendEmailParams } from '../types';
+import config from 'config';
+import mail from '@sendgrid/mail';
+
+export default class SendGridService {
+  public static initializeService() {
+    const sendgridAPIKey:string = config.get('sendgridAPIKey'); // Todo: here use configuration service to get these details.
+    if(sendgridAPIKey) {
+      mail.setApiKey(sendgridAPIKey);
+    }
+  }
+  public static async sendEmail(params: SendEmailParams): Promise<void> {
+    const msg: any = {
+      to: params.to,
+      from: {
+        email: params.from,
+        name: params.fromName,
+      }
+    };
+    msg.templateId = params.templateId;
+    msg.dynamic_template_data = params.templateData;
+    msg.dynamic_template_data = Object.assign(msg.dynamic_template_data, {
+      domain: config.get('webAppHost'),
+    });
+    try {
+      await mail.send(msg);
+    } catch(error) {
+      // Todo: Use logger Service to log these values
+      console.log(error.message);
```

Comment:
We should throw error so the caller knows something went wrong and can take corrective action.

We should create ThirdPartyServiceError and throw its instance from here